### PR TITLE
chore: Set project-wide config for CLIRR version compatibility checks

### PR DESCRIPTION
**Pull request by hessjcg**
Update the maven configuration so that the CLIRR plugin only checks classes that are documented
to be part of our public API.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -240,6 +240,24 @@
...
       </plugin>
+
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>clirr-maven-plugin</artifactId>
+        <configuration>
+          <includes>com/google/cloud/sql/CredentialFactory</includes>
+          <includes>com/google/cloud/sql/core/CoreSocketFactory</includes>
+          <includes>com/google/cloud/sql/mariadb/SocketFactory</includes>
+          <includes>com/google/cloud/sql/mariadb/SocketFactory</includes>
```
Comment (ttosta-google) [inline pom.xml:251]: This line is the same as the one above.

**Review by ttosta-google — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (hessjcg) [reply pom.xml]: Good catch. Fixed.

**Comment by hessjcg** [comment]
Update the maven configuration so that the CLIRR plugin only checks classes that are documented
to be part of our public API.

File: pom.xml (hunk 1 above)
Comment (ttosta-google) [reply pom.xml]: LGTM

**Pull request merged**